intercept/src/dispatch.cpp

```diff
@@ -4766,12 +4785,19 @@ CL_API_ENTRY cl_int CL_API_CALL CLIRN(clEnqueueNDRangeKernel)(
 {
     CLIntercept*    pIntercept = GetIntercept();
 
+    // In case we want to dump a replayble kernel by kernel name, we only do this on the first enqueue
+    static bool hasDumpedBufferByName = false;
+    static bool hasDumpedValidationBufferByName = false;
+    static bool hasDumpedImageByName = false;
+    static bool hasDumpedValidationImageByName = false;
```

Comment:
Possible, I can just add a mutex around these. Putting them in the CLIntercept class also works, but given that the scope of these variables is local to this function, this would be one step forward one step backwards. So, I'll following your decision!